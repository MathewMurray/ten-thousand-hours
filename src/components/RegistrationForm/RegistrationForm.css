.full_name, .user_name, .password {
    text-align: center;
    
}
#RegistrationForm__full_name{
    padding: 1px .5rem 4px 4rem;
    margin: 3px .5rem;
    border: 1px solid #888383;
}

#RegistrationForm__user_name{
    padding: 1px .5rem 4px 4rem;
    margin: 4px 1rem;
    border: 1px solid #888383;
}
#registrationForm__password{
    padding: 1px .5rem 4px 4rem;
    margin: 4px 1rem;
    border: 1px solid #888383;
}
p.red{
    text-align: center;
}
