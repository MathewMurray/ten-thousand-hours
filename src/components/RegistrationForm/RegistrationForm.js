import React, {Component} from 'react'
import {Button, Input, Required} from '../Utils/Utils'

export default class RegistrationForm extends Component{
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = {error:null}

    handleSubmit = ev => {
        ev.preventDefault()
        const {full_name,user_name,password} = ev.target

        console.log('registration form submitted')
        console.log(full_name,user_name,password)

        full_name.value = ''
        user_name.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
    }

    render(){
        const {error} = this.state
        return(
            <form className='RegistrationForm' onSubmit={this.handleSubmit}>
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className='full_name'>
                    <label htmlFor='RegistrationForm__full_name'>Full Name<Required /></label>
                    <Input name='full_name' type='text' Required id='RegistrationForm__full_name'></Input>
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm__user_name'>User Name<Required /></label>
                    <Input name='user_name' type='text' required id='RegistrationForm__user_name'></Input>
                </div>
                <div className='password'>
                    <label htmlFor='RegistrationForm__password'>Password <Required /></label>
                    <Input name='password' type='password' required id='registrationForm__password'></Input>
                </div>
                <Button type='submit'>Register</Button>
            </form>
        )
    }
}