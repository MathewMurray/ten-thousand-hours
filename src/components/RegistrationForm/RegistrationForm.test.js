import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationForm from './RegistrationForm'

describe(`RegistrationForm component`, () => {
    it('renders the registration form', () => {
        const wrapper = shallow(<RegistrationForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})