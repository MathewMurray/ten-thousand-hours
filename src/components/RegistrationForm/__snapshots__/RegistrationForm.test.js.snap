// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders the registration form 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full Name 
      <Required />
    </label>
    <Input
      id="RegistrationForm__full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      Username 
      <Required />
    </label>
    <Input
      id="RegistrationForm__user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password 
      <Required />
    </label>
    <Input
      id="registrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
</form>
`;
